shared/actions/fs/common.native.js

```diff
@@ -1,33 +1,25 @@
 // @flow
+import * as I from 'immutable'
+import * as Types from '../../constants/types/fs'
 import * as Saga from '../../util/saga'
 import * as FsGen from '../fs-gen'
 import {putActionIfOnPath, navigateAppend} from '../route-tree'
 
-export const share = (action: FsGen.SharePayload) =>
+const getTransferPopupAction = (path: Types.Path, routePath?: I.List<string>) =>
   Saga.put(
-    putActionIfOnPath(
-      action.payload.routePath,
-      navigateAppend([
-        {
-          props: {path: action.payload.path, isShare: true},
-          selected: 'pathItemAction',
-        },
-      ])
-    )
+    routePath
+      ? putActionIfOnPath(routePath, navigateAppend([{props: {path}, selected: 'transferPopup'}]))
+      : navigateAppend([{props: {path}, selected: 'transferPopup'}])
   )
 
-export function* save(action: FsGen.SavePayload): Saga.SagaGenerator<any, any> {
-  const {path, routePath} = action.payload
-  Saga.put(FsGen.createDownload({path, intent: 'camera-roll'}))
-  Saga.put(
-    putActionIfOnPath(
-      routePath,
-      navigateAppend([
-        {
-          props: {path, isShare: true},
-          selected: 'transferPopup',
-        },
-      ])
-    )
-  )
-}
+export const shareNative = ({payload: {path, routePath}}: FsGen.ShareNativePayload) =>
+  Saga.sequentially([
+    Saga.put(FsGen.createDownload({intent: 'share', path})),
+    getTransferPopupAction(path, routePath),
+  ])
+
+export const saveMedia = ({payload: {path, routePath}}: FsGen.SaveMediaPayload) =>
+  Saga.sequentially([
+    Saga.put(FsGen.createDownload({path, intent: 'camera-roll'})),
```

Comment:
please keep `intent` and `path` in the same order for the various `FsGen.createDownload` calls.